client.cpp

```diff
@@ -96,6 +96,28 @@ static void segfault(int sig, siginfo_t* info, void* unused) {
     raise(SIGSEGV);
 }
 
+void maybe_copy_unified_arg(const int index, void* arg, enum cudaMemcpyKind kind)
```

Comment:
Make this return -1 if any cudaMemcpy fails? We probably want to return early in the generated rpc's if the memcpy's explicitly fail.